### PR TITLE
WIP: Migrate eslint config

### PR DESCRIPTION
**Pull request by bzang**

### CODE REVIEW

**Comment by adamweeks** [comment]
Deploy preview ready!

Built with commit d9a0e9927df3f791483e499b1a0c54299dc8810d

https://deploy-preview-586--hopeful-hamilton-4090b9.netlify.com

File: packages/node_modules/@ciscospark/widget-meet/src/container.js [hunk 1]
```diff
@@ -150,13 +150,13 @@ export class MeetWidget extends Component {
       this.props.checkWebRTCSupport(sparkInstance);
     }
     if (isWebRTCSupported) {
-      this.setup(this.props);
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-meet/src/container.js]: What rule stops us from using `this`?

Comment (bzang) [reply packages/node_modules/@ciscospark/widget-meet/src/container.js]: The setup method doesn’t actually use this, so I moved it to a static method which means we can’t reference it with this.setup. Nothing stops us from using this.

File: packages/node_modules/@ciscospark/react-component-title-bar/src/index.js [hunk 2]
```diff
@@ -23,7 +22,7 @@ function TitleBar({avatarId, children, name}) {
   return (
     <div className={classNames('ciscospark-title-bar', styles.titleBar)}>
       <div className={classNames('ciscospark-avatar-container', styles.avatarContainer)}>
-        <PresenceAvatar avatarId={avatarId} iconSize={'30px'} name={name} />
+        <PresenceAvatar avatarId={avatarId} iconSize="30px" name={name} />
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/react-component-title-bar/src/index.js]: The react rule doesn't require these strings any more?

File: packages/node_modules/@ciscospark/react-component-typing-avatar/src/index.js [hunk 3]
```diff
@@ -1,11 +1,11 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import PresenceAvatar from '@ciscospark/react-container-presence-avatar';
-import classNames from 'classnames';
 
-import Avatar from '@ciscospark/react-component-avatar';
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/react-component-typing-avatar/src/index.js]: Not sure how it let me keep that in

File: packages/node_modules/@ciscospark/react-container-read-receipts/src/index.js [hunk 4]
```diff
@@ -37,13 +37,12 @@ export class ReadReceipts extends Component {
   render() {
     const {users, totalCount} = this.props.readReceipts;
     const readReceipts = users.map((user) =>
-      <TypingAvatar
+      (<TypingAvatar
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/react-container-read-receipts/src/index.js]: What does wrapping in parens give us?

Comment (bzang) [reply packages/node_modules/@ciscospark/react-container-read-receipts/src/index.js]: This was enabled https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-wrap-multilines.md. Gives better readability for multi-line jsx

File: packages/node_modules/@ciscospark/react-component-title-bar/src/index.js (hunk 2 above)
Comment (bzang) [reply packages/node_modules/@ciscospark/react-component-title-bar/src/index.js]: Yea. That was an oversight on my part. I think we should enable it as a warn, but as is we tend to ignore the rule and get around it anyways. What do you think? Is there another way we can encourage or enforce not using raw strings when it’s displayed text?

File: packages/node_modules/@ciscospark/react-container-read-receipts/src/index.js (hunk 4 above)
Comment (adamweeks) [reply packages/node_modules/@ciscospark/react-container-read-receipts/src/index.js]: img

No more eslint disable!

File: packages/node_modules/@ciscospark/react-component-title-bar/src/index.js (hunk 2 above)
Comment (adamweeks) [reply packages/node_modules/@ciscospark/react-component-title-bar/src/index.js]: I haven't ever gotten around it, but I do like the idea of it. Although now that I'm reading the rule: https://github.com/yannickcr/eslint-plugin-react/blob/HEAD/docs/rules/jsx-no-literals.md I think this is just for string within a component and not params.

**Pull request merged**